client/src/components/UserDashBoard.jsx

```diff
@@ -1,16 +1,121 @@
 const React = require('react');
 const UserDashBoardKeywordsEntryBar = require('./UserDashBoardKeywordsEntryBar.jsx');
+const UserDashBoardMonitoredWordResult = require('./UserDashBoardMonitoredWordResult.jsx');
 
 class UserDashBoard extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isFetching: true,
+      monitoringResults: this.props.monitoringResults,
+      errorMessage: ''
+    };
+
+    this.handleAddMonitoredWords = this.handleAddMonitoredWords.bind(this);
+    this.handleRemoveMonitoredWords = this.handleRemoveMonitoredWords.bind(this);
+  }
+  componentDidMount() { 
+    // Retrieve user monitored keywords and associated bills to populate the dashboard
+    $.get('user/' + this.props.username + '/keywords')
+      .done(data => this.setState({
+        isFetching: false,
+        monitoringResults: data,
+        'errorMessage': ''
+      }))
+      .fail(error => this.setState({
+        isFetching: false,
+        monitoringResults: [],
+        errorMessage: error.status + '-' + error.statusText
+      }));
+  }
+
+  handleAddMonitoredWords(words) {
+    // PUT message to server to add a new monitored word(s) string
+    $.ajax('user/' + this.props.username + '/keywords', {
+      method: 'PUT',
+      context: this,
+      data: {
+        keyword: words
+      },
+      dataType: 'json'
+    })
+      .done(function(data) {
+
+        let revisedMonitoringResults = this.state.monitoringResults.slice();
+        revisedMonitoringResults.unshift(data);
+
+        console.log(revisedMonitoringResults);
+        // Add the new entry into our dashboard, along with the monitored keywords 'hits' returned
+        // by our server
+        if (data !== undefined) {
+          this.setState({
+            monitoringResults: revisedMonitoringResults
+          });
+        }
+      });
+  }
+
+  handleRemoveMonitoredWords(words) {
+    //DELETE message to server to remove a monitored word(s) string
+    $.ajax('user/' + this.props.username + '/keywords', {
```

Comment:
Here we can also send you an updated data with the following data type:

{keywords: {"keyword1": {keyword: "keyword1",
associatedKeywords: [...,...,],
bills: [id1, id2, id3]}
}